test/lib/dates.lib.test.js

```diff
@@ -86,4 +86,100 @@ describe('Dates lib', () => {
       expect(result).to.be.true()
     })
   })
+
+  describe('cycleDueDate', () => {
+    it('should return the due date of the summer cycle with summer as true', () => {
+      const summerDueDate = new Date(new Date().getFullYear() + 1, 10, 28)
+      const result = DateLib.cycleDueDate(true)
+
+      expect(result).to.equal(summerDueDate)
+    })
+
+    it('should return the due date of the winter and all year cycle  with summer as false', () => {
+      const allYearDueDate = new Date(new Date().getFullYear() + 1, 3, 28)
+      const result = DateLib.cycleDueDate(false)
+
+      expect(result).to.equal(allYearDueDate)
+    })
+  })
```

Comment:
Just highlighting this one, but the example applies to the rest of the new date tests.

First, the aim is that the test output we see reads as much as possible as a comprehensive sentence.

So, currently these will read as

```text
cycleDueDate
  [it] should return the due date of the summer cycle with summer as true
  [it] should return the due date of the winter and all year cycle with summer as false
```

Ideally, I'd be aiming for something like this.

```text
#cycleDueDate
  when the requested cycle is "summer"
    it] should return the due date for the current summer cycle
  when the requested cycle is not "summer" (winter and all year)
    [it] should return the due date of the current winter and all year cycle
```

The next aim is to follow the AAA pattern (Arrange, Act, Assert) with the tests. To make this clear we try to always ensure that any setup (Arrange) is done outside of the `it()` in a `beforeEach()`. The first line in the `it()` then becomes our Act, followed by any assertions.

So, my suggested changes for this would be
```javascript
  // You'll want to place these at the top of the file after the initial describe()
  let expectedDate
  let summer

  describe('#cycleDueDate', () => {
    describe('when the requested cycle is "summer"', () => {
      beforeEach(() => {
        summer = true

        expectedDate = new Date(new Date().getFullYear() + 1, 10, 28)
      })

      it('should return the due date for the current summer cycle', () => {
        const result = DateLib.cycleDueDate(summer)
  
        expect(result).to.equal(expectedDate)
      })
    })
    
    describe('when the requested cycle is "summer"', () => {
      beforeEach(() => {
        summer = false

        expectedDate = new Date(new Date().getFullYear() + 1, 3, 28)
      })
    })

    it('should return the due date of the current winter and all year cycle', () => {
      const result = DateLib.cycleDueDate(summer)

      expect(result).to.equal(expectedDate)
    })
  })
```